packages/aws-cdk-lib/aws-codedeploy/lib/private/utils.ts

```diff
@@ -31,12 +31,16 @@ export function arnForDeploymentConfig(name: string, resource?: IResource): stri
   });
 }
 
-export function renderAlarmConfiguration(alarms: cloudwatch.IAlarm[], ignorePollAlarmFailure: boolean | undefined, removeAlarms = true):
-CfnDeploymentGroup.AlarmConfigurationProperty | undefined {
+export function renderAlarmConfiguration(
+  alarms: cloudwatch.IAlarm[],
+  ignorePollAlarmFailure?: boolean,
+  ignoreAlarmConfiguration = false,
+  removeAlarms = true,
```

Comment:
I believe the new property should be at the end. Since this is an exported function, someone could be using it and now their third parameter would be referencing another argument.

Reply:
Funny enough, that's what I had originally. There was a request to flip it [here](https://github.com/aws/aws-cdk/pull/26957#discussion_r1348519360) for readability, but to your point usability should probably supersede.

Reply:
@vinayak-kukreja is right, we cannot flip it because it would be a breaking change to flip. However, I have an alternate solution: we deprecate `renderAlarmConfiguration` entirely and build `renderAlarmConfigurationV2`.  

It would look like this:

```ts
export function renderAlarmConfiguration(alarms: cloudwatch.IAlarm[], ignorePollAlarmFailure: boolean | undefined, removeAlarms = true) {
  return renderAlarmConfigurationV2({ alarms, ignorePollAlarmFailure, removeAlarms });
}

export interface renderAlarmConfigProps {
  readonly alarms: cloudwatch.IAlarm[],
  readonly ignorePollAlarmFailure?: boolean,
  /** default true */
  readonly removeAlarms?: boolean,
  // your new prop
}

export function renderAlarmConfigurationV2(props: renderAlarmConfigProps) {
  // the original implementation plus your new prop
}
```

This would mean updating all the places we use `renderAlarmConfiguration` to `renderAlarmConfigurationV2`, but I think this is the correct path to go down.